fs-api-blockio/src-linux/System/FS/BlockIO/Async.hs

```diff
@@ -97,3 +101,29 @@ ioopConv (IOOpWrite h off buf bufOff c) = handleFd h >>= \fd ->
 -- would require a change in @fs-api@. See [fs-sim#49].
 handleFd :: Handle HandleIO -> IO Fd
 handleFd h = withOpenHandle "submitIO" (handleRaw h) pure
+
+-- | Heterogeneous blend of `V.zipWithM` and `VU.zipWithM`
+--
+-- The @vector@ package does not provide functions that take distinct vector
+-- containers as arguments, so we write it by hand to prevent having to convert
+-- one vector type to the other.
+hzipWithM ::
+     (IOOp IO HandleIO -> I.IOResult -> IO IOResult)
+  -> V.Vector (IOOp IO HandleIO)
+  -> VU.Vector I.IOResult
+  -> IO (VU.Vector IOResult)
```

Comment:
The implementation is also not a based on bundles, which is probably sensible here.

I like that you split it out as a separate function that has a familiar interface. I think it would be even nicer if it was a little more polymorphic, i.e. took a function `a -> b -> m c`.